provider/oci/common_test.go

```diff
@@ -1,19 +1,144 @@
 package oci_test
 
 import (
+	"fmt"
+	"time"
+
+	gitjujutesting "github.com/juju/testing"
+	"github.com/juju/utils/arch"
+	"github.com/juju/version"
+
 	gomock "github.com/golang/mock/gomock"
 	ocitesting "github.com/juju/juju/provider/oci/testing"
 	jujutesting "github.com/juju/juju/testing"
 	jc "github.com/juju/testing/checkers"
 	gc "gopkg.in/check.v1"
 
 	ociCore "github.com/oracle/oci-go-sdk/core"
+	ociIdentity "github.com/oracle/oci-go-sdk/identity"
 
 	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
 	"github.com/juju/juju/environs/tags"
 	"github.com/juju/juju/provider/oci"
+	"github.com/juju/juju/tools"
 )
 
+var clk = gitjujutesting.NewClock(time.Time{})
+var advancingClock = gitjujutesting.AutoAdvancingClock{clk, clk.Advance}
+
+func makeToolsList(series string) tools.List {
+	var toolsVersion version.Binary
+	toolsVersion.Number = version.MustParse("2.4.0")
+	toolsVersion.Arch = arch.AMD64
+	toolsVersion.Series = series
+	return tools.List{{
+		Version: toolsVersion,
+		URL:     fmt.Sprintf("http://example.com/tools/juju-%s.tgz", toolsVersion),
+		SHA256:  "1234567890abcdef",
+		Size:    1024,
+	}}
+}
+
+func makeListSecurityListsRequestResponse(seclistDetails []ociCore.SecurityList) (ociCore.ListSecurityListsRequest, ociCore.ListSecurityListsResponse) {
+	if len(seclistDetails) == 0 {
+		return ociCore.ListSecurityListsRequest{}, ociCore.ListSecurityListsResponse{}
+	}
+
+	compartment := seclistDetails[0].CompartmentId
+	vcnId := seclistDetails[0].VcnId
+	request := ociCore.ListSecurityListsRequest{
+		CompartmentId: compartment,
+		VcnId:         vcnId,
+	}
+	response := ociCore.ListSecurityListsResponse{
+		Items: seclistDetails,
+	}
+
+	return request, response
+}
+
+func makeListInternetGatewaysRequestResponse(gwDetails []ociCore.InternetGateway) (ociCore.ListInternetGatewaysRequest, ociCore.ListInternetGatewaysResponse) {
+	if len(gwDetails) == 0 {
+		return ociCore.ListInternetGatewaysRequest{}, ociCore.ListInternetGatewaysResponse{}
+	}
+
+	compartment := gwDetails[0].CompartmentId
+	vcnId := gwDetails[0].VcnId
+	request := ociCore.ListInternetGatewaysRequest{
+		CompartmentId: compartment,
+		VcnId:         vcnId,
+	}
+	response := ociCore.ListInternetGatewaysResponse{
+		Items: gwDetails,
+	}
+	// for _, val := range gwDetails {
```

Comment:
commented out code should usually be removed, unless it is being used in the next patch.